PacManGame/ConsoleUserInput.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace PacManGame
+{
+  public class ConsoleUserInput : IUserInput
+  {
+    public Direction GetUserInput()
+    {
```

Comment:
This method is more like a converter. Think of input/output interfaces like a way of talking in different languages. Say all your interface needs to do is greet(), you could have one for italian so it would be greet() { return ciao } or spanish greet() { return ola } . The userninput just needs a way to read and the output needs a way to write. Then when a console input class is used the read() method would call console.readLine (or read key) and use that implementation of a way to read what a user has inputed